### DISCUSSION TITLE
XCode 15 Strings Catalog

### DISCUSSION THREAD

**Discussion by Kondamon** [SwiftGen/SwiftGen#1071] [Q&A] [answered]
Does anyone have already made any experiences regarding String catalog?

Do you recommend in the long run to just skip Swiftgen for enum L10n generation and let Xcode automatically extract Strings to the new string catalog?

**Comment by neilkachu** [comment, marked as answer]
Hi,

I have been making some progress on this as I already go from strings>enum and wanted to move all strings to string catalog as I have 70 targets and it would make spotting completeness better.

As xcstrings files are json Ive made a simple template to convert json>strings then the run standard strings> enum in the yaml file

The only issue I found was as the available methods for the json parser seems to treat strings with /n in them as paragraphs (unlike the .translation property on the strings parser) Ive had to pre-convert all these to {newline} and process them back later.

This is basically it, Im only interested in the en localisation as these are always complete.

``` Generated using SwiftGen — https://github.com/SwiftGen/SwiftGen

{% if files %}
{% macro fileBlock file %}
  {% call documentBlock file file.document %}
{% endmacro %}

{# process the file #}
{% macro documentBlock file document %}
  
  {% if document.metadata.type == "Dictionary" %}
	{% for key,value in document.metadata.properties %}
		{% call propertyBlock key value document.data %}
	{% endfor %}
 {% endif %}
{% endmacro %}

{# process the root dictionary #}
{% macro propertyBlock key metadata data %}
  {% set propertyName %}{{key|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}}{% endset %}
	{% if propertyName == "strings" %}
		    {% for propertyKey in data[key] %}
				{% set propertyValue %}{{data[key][propertyKey].localizations.en.stringUnit.value}}{% endset %}
				{% if propertyValue %}
"{{propertyKey}}"="{{propertyValue}}";
				{% endif %}
			{% endfor %}
	  {% else %}
	  {% endif %}
{% endmacro %}

{% if files.count > 1 or param.forceFileNameEnum %}
{% for file in files %}
	{% filter indent:2," ",true %}{% call fileBlock file %}{% endfilter %}
{% endfor %}
{% else %}
{% call fileBlock files.first %}
{% endif %}

{% else %}
// No files found
{% endif %}```
